apis/v1beta1/collector_webhook.go

```diff
@@ -98,7 +100,7 @@ func (c CollectorWebhook) Default(_ context.Context, obj runtime.Object) error {
 		trueVal := true
 		otelcol.Spec.NetworkPolicy.Enabled = &trueVal
 	}
-	return otelcol.Spec.Config.ApplyDefaults(c.logger)
+	return otelcol.Spec.Config.ApplyDefaults(c.logger, c.recorder, otelcol)
```

Comment:
why are we passing otelcol down to itself?

Reply:
I also think another option here would be to have the ApplyDefaults method return a set of events which would allow us to test this more effectively. LMK what you think about that approach. this would allow us to not need to drill down the recorder and separate runtime.Obj reference.